integration_test/multicall.test.ts

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+import { getL2Network } from '../src/lib/dataEntities/networks'
+import { providers } from 'ethers'
+import { mock, when, anything, instance, deepEqual } from 'ts-mockito'
+import { expect } from 'chai'
+
+import { MultiCaller } from '../src'
+
+describe('Multicall events', () => {
+  const createProviderMock = async (networkChoiceOverride?: number) => {
+    const l2Network = await getL2Network(networkChoiceOverride || 42161)
+
+    const l2ProviderMock = mock(providers.JsonRpcProvider)
+    const latestBlock = l2Network.nitroGenesisBlock + 1000
+    when(l2ProviderMock.getBlockNumber()).thenResolve(latestBlock)
+    when(l2ProviderMock.getNetwork()).thenResolve({
+      chainId: l2Network.chainID,
+    } as any)
+    when(l2ProviderMock._isProvider).thenReturn(true)
+    when(l2ProviderMock.getLogs(anything())).thenResolve([])
+    // Maker
+    when(
+      l2ProviderMock.call(
+        deepEqual({
+          data: '0xbce38bd7000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000009f8f72aa9304c8b593d555f12ef6589cc3a579a20000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000406fdde03000000000000000000000000000000000000000000000000000000000000000000000000000000009f8f72aa9304c8b593d555f12ef6589cc3a579a20000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000495d89b4100000000000000000000000000000000000000000000000000000000',
+          to: '0x108B25170319f38DbED14cA9716C54E5D1FF4623',
+        }),
+        undefined
+      )
+    ).thenResolve(
+      '0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000204d616b65720000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000204d4b520000000000000000000000000000000000000000000000000000000000'
```

Comment:
Can you add a comment of the steps you took to produce these arguments, might be helpful for a future dev when changing these tests